docs/data-migration.md

```diff
@@ -0,0 +1,105 @@
+migration and then the `lnd` configuration, so keep that string somewhere
+(should be something with the format of `postgres://xx:yy@localhost:5432/zz`).
+
+No additional steps are required to prepare the Postgres database for the data
+migration. The migration tool will create the database schema automatically, so
+no DDL scripts need to be run in advance.
+
+## Prepare the source database
+
+Assuming we want to migrate the database state from the pre-0.16.0 individual
+`bbolt` based `*.db` files to a remote database, we first need to make sure the
+source files are in the correct state.
+
+The following steps should be performed *before* running the data migration:
+1. Stop `lnd`
+2. Upgrade the `lnd` binary to the latest version (e.g. `v0.16.0-beta` or later)
+3. Make sure to add config options like `gc-canceled-invoices-on-startup=true`
+   and `db.bolt.auto-compact=true` to your `lnd.conf` to optimize the source
+   database size by removing canceled invoices and compacting it on startup.
+4. Start `lnd` normally, using the flags mentioned above but not yet changing
+   any database backend related configuration options. Check the log that the
+   database schema was migrated successfully, for example: `Checking for
+   schema update: latest_version=XX, db_version=XX`
+5. Remove any data from the source database that you can. The fewer entries are
+   in the source database, the quicker the migration will complete. For example
+   failed payments (or their failed HTLC attempts) can be removed with
+   `lncli deletepayments`.
+6. Stop `lnd` again and make sure it isn't started again by accident during the
+   data migration (e.g. disable any `systemd` or other scripts that start/stop
+   `lnd`).
+
+NOTE: If you were using the experimental `etcd` cluster mode that was introduced
+in `lnd v0.13.0` it is highly recommended starting a fresh node. While the data
+can in theory be migrated from the partial/mixed `etcd` and `bbolt` based state
+the migration tool does not support it.
+
+## Run the migration
+
+Depending on the destination database type, run the migration with a command
+similar to one of the following examples:
+
+**Example: Migrate from `bbolt` to `etcd`:**
+
+```shell
+⛰  lndinit -v migrate-db \
+      --source.bolt.data-dir /home/myuser/.lnd/data \
+      --source.bolt.tower-dir /home/myuser/.lnd/watchtower \
+      --dest.backend etcd \
+      --dest.etcd.host=my-etcd-cluster-address:2379
+```
+If you weren't running a watchtower server, you can remove the line with
+`--source.bolt.tower-dir`.
+
+**Example: Migrate from `bbolt` to `postgres`:**
+
+```shell
+⛰  lndinit -v migrate-db \
+      --source.bolt.data-dir /home/myuser/.lnd/data \
+      --source.bolt.tower-dir /home/myuser/.lnd/data/watchtower \
+      --dest.backend postgres \
+      --dest.postgres.dsn=postgres://postgres:postgres@localhost:5432/postgres
+```
+
+If you weren't running a watchtower server, you can remove the line with
+`--source.bolt.tower-dir`.
```

Comment:
Should end with a section advising user to run the migrated node with the updated db config settings